src/main/java/ru/yandex/clickhouse/ClickHouseDatabaseMetadata.java

```diff
@@ -97,17 +101,34 @@ public String getDriverName() throws SQLException {
 
     @Override
     public String getDriverVersion() throws SQLException {
-        return "0.1";
+        String driverVersion = getClass().getPackage().getImplementationVersion();
+        return driverVersion != null ? driverVersion : "0.1";
     }
 
     @Override
     public int getDriverMajorVersion() {
-        return 0;
+        String v;
+        try {
+            v = getDriverVersion();
+        } catch (SQLException sqle) {
+            log.warn("Error determining driver major version", sqle);
+            return 0;
+        }
+        Matcher m = VERSION_NUMBER_PATTERN.matcher(v);
+        return m.matches() ? Integer.parseInt(m.group(1)) : 0;
     }
 
     @Override
     public int getDriverMinorVersion() {
-        return 1;
+        String v;
```

Comment:
I suppose the common part from these methods may be extracted to something returning either array or version object to make it clearer and avoid duplications

Reply:
good suggestion!